File: android/src/main/kotlin/com/bowvie/flutter_on_device_ai/FlutterOnDeviceAiPlugin.kt
package com.bowvie.flutter_on_device_ai

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** FlutterOnDeviceAiPlugin */
class FlutterOnDeviceAiPlugin : FlutterPlugin {
    private var bridge: OnDeviceAiBridge? = null

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        val messenger = flutterPluginBinding.binaryMessenger
        bridge = OnDeviceAiBridge().also {
            OnDeviceAiHostApi.setUp(messenger, it)
            it.registerStreamHandler(messenger)
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        OnDeviceAiHostApi.setUp(binding.binaryMessenger, null)
        bridge?.close()
        bridge = null
    }
}

File: ios/flutter_on_device_ai/Sources/flutter_on_device_ai/FlutterOnDeviceAiPlugin.swift
import Flutter

public class FlutterOnDeviceAiPlugin: NSObject, FlutterPlugin {
  private let bridge = OnDeviceAiBridge()

  public static func register(with registrar: FlutterPluginRegistrar) {
    let instance = FlutterOnDeviceAiPlugin()
    OnDeviceAiHostApiSetup.setUp(
      binaryMessenger: registrar.messenger(),
      api: instance.bridge
    )
    instance.bridge.registerStreamHandler(with: registrar.messenger())
  }
}
